src/utils.js

```diff
@@ -176,3 +176,25 @@ export function getRecordStandardIdentifiers(record) {
 export function clone(o) {
 	return JSON.parse(JSON.stringify(o));
 }
+
+export function updateField001ToParamId(id, record) {
+	let fields = record.fields.filter(field => filterFieldsByPattern(field, /^001$/));
+
+	if (fields.length === 0) {
+		return record.fields.push({tag: '001', value: id});
+	}
+
+	return record.get(/^001$/).map(field => {
```

Comment:
- The fields have already been retrieved to `fields`. Use that variable.
- The ID must be in Aleph's id format. Nick `toAlephId` & `fromAlephId(` from [OAI-PMH](https://github.com/NatLibFi/melinda-oai-pmh-provider/blob/dev/src/app/middleware/db/record.js) and add to `utils.js` (I'll change OAI-PMH later to use them from commons).